src/vs/workbench/contrib/search/browser/replaceService.ts

```diff
@@ -109,7 +110,20 @@ export class ReplaceService implements IReplaceService {
 		const edits = this.createEdits(arg, resource);
 		await this.bulkEditorService.apply(edits, { progress });
 
-		return Promises.settled(edits.map(async e => this.textFileService.files.get(e.resource)?.save({ source: ReplaceService.REPLACE_SAVE_SOURCE })));
+		const rawTextPromises = edits.map(async e => {
+			if (e.resource.scheme === network.Schemas.vscodeNotebookCell) {
+				const notebookResource = CellUri.parse(e.resource)?.notebook;
+				if (notebookResource) {
+					return this.editorService.save([...this.editorService.findEditors(notebookResource)]);
```

Comment:
Maybe we should be doing this at the model level as well like for text files? I don't know which service would do that.

Reply:
I was under the impression that saving the text models happened here https://github.com/microsoft/vscode/blob/349b100fb66797133a04b2c2edef36c044b33b8d/src/vs/workbench/contrib/search/browser/replaceService.ts#L122 (even if it's unsaved)